File: packages/nodejs/src/auth/api-key.ts
import type { Metadata } from '@grpc/grpc-js';
import type { VMXClient } from '../client';
import type { VMXClientAuthProvider } from './types';

export type VMXClientAPIKeyOptions = {
  apiKey: string;
};

export class VMXClientAPIKey implements VMXClientAuthProvider {
  constructor(private readonly options: VMXClientAPIKeyOptions) {}

  public async injectCredentials(_client: VMXClient, grpcMetadata: Metadata): Promise<void> {
    grpcMetadata.add('x-api-key', this.options.apiKey);
  }
}

File: packages/python/vmxai/auth/api_key.py
from typing import TYPE_CHECKING

from vmxai.auth.provider import VMXClientAuthProvider

if TYPE_CHECKING:
    from vmxai.client import VMXClient


class VMXClientAPIKey(VMXClientAuthProvider):
    def __init__(self, api_key: str):
        self.api_key = api_key

    def inject_credentials(self, client: "VMXClient", grpc_metadata: list[tuple[str, str]]) -> tuple[tuple[str, str]]:
        grpc_metadata.append(("x-api-key", self.api_key))
        return grpc_metadata
